src/Microsoft.AspNet.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs

```diff
@@ -96,29 +96,23 @@ internal static void AddRazorViewEngineServices(IServiceCollection services)
                 return new DefaultChunkTreeCache(cachedFileProvider.Options.FileProvider);
             }));
 
-            // The host is designed to be discarded after consumption and is very inexpensive to initialize.
-            services.TryAddTransient<IMvcRazorHost, MvcRazorHost>();
-
             // Caches compilation artifacts across the lifetime of the application.
             services.TryAddSingleton<ICompilerCacheProvider, DefaultCompilerCacheProvider>();
 
-            // This caches compilation related details that are valid across the lifetime of the application
-            // and is required to be a singleton.
+            // This caches compilation related details that are valid across the lifetime of the application.
             services.TryAddSingleton<ICompilationService, RoslynCompilationService>();
 
-            // Both the compiler cache and roslyn compilation service hold on the compilation related
-            // caches. RazorCompilation service is just an adapter service, and it is transient to ensure
-            // the IMvcRazorHost dependency does not maintain state.
+            // In the default scenario the following services are singleton by virtue of being initialized as part of
+            // creating the singleton RazorViewEngine instance.
```

Comment:
Question: If these are effectively singletons why aren't they added as singletons? Does something else reference them in a way that requires additional instances?